File: Venda-distribuida-java/src/main/java/br/ufc/venda/client/esqueleto/EsqueletoBroadcastVenda.java
package br.ufc.venda.client.esqueleto;

import com.google.gson.Gson;

import br.ufc.venda.client.dao.VendaDAO;
import br.ufc.venda.model.Resposta;
import br.ufc.venda.model.Venda;

public class EsqueletoBroadcastVenda {
	
	private VendaDAO vdao;
	private Gson gson;
	
	public EsqueletoBroadcastVenda(){
		vdao = new VendaDAO();
		gson = new Gson();
	}
	
	public String refresh(String args){
		Venda venda = gson.fromJson(args, Venda.class);
		Resposta resposta = new Resposta();
		if(vdao.salvar(venda)){
			resposta.setCodigo(Resposta.SUCCESS);
			resposta.setResposta("true");
		}else{
			resposta.setCodigo(Resposta.ERROR);
			resposta.setResposta("false");
		}
		return gson.toJson(resposta);
	}
	
	public String rollback(String args){
		Venda venda = gson.fromJson(args, Venda.class);
		Resposta resposta = new Resposta();
		if(vdao.excluir(venda)){
			resposta.setCodigo(Resposta.SUCCESS);
			resposta.setResposta("true");
		}else{
			resposta.setCodigo(Resposta.ERROR);
			resposta.setResposta("false");
		}
		return gson.toJson(resposta);
	}
	
	
}

File: Venda-distribuida-python/esqueleto/EsqueletoBroadcastVenda.py
#-*-coding:utf-8 -*-

import json

from dao.VendaDAO import VendaDAO

class EsqueletoBroadcastVenda():

    __SUCCESS = 200
    __ERROR = 500

    def __init__(self):
        self.vdao = VendaDAO()

    def refresh(self, args):
        venda = json.loads(args)
        resposta = None
        if self.vdao.salvar(venda):
            resposta = {"codigo":self.__SUCCESS, "reposta":"true"}
        else:
            resposta = {"codigo": self.__ERROR, "reposta": "false"}
        return json.dump(resposta)

    def rollback(self,args):
        venda = json.loads(args)
        resposta = None
        if self.vdao.excluir(venda):
            resposta = {"codigo": self.__SUCCESS, "reposta": "true"}
        else:
            resposta = {"codigo": self.__ERROR, "reposta": "false"}
        return json.dump(resposta)
